docs/website/docs/developers/design-docs/hip-hal-driver.md

```diff
@@ -0,0 +1,263 @@
+---
+hide:
+  - tags
+tags:
+  - GPU
+  - HIP
```

Comment:
I'm not sure how much use we're getting out of tags, but we could merge `HIP` and `ROCm` into just `AMD`, or keep both. Fine to leave this as-is, just FYI for now...

Here's a page showing all tags and associated pages: https://iree.dev/community/tags/

Website search also shows tags: 
![image](https://github.com/openxla/iree/assets/4010439/11b179f8-67c8-41f9-bcba-1061eeeb60d4)